chain-signatures/node/src/protocol/state.rs

```diff
@@ -77,18 +79,47 @@ pub struct RunningState {
 
 pub struct ResharingState {
     pub me: Participant,
-    pub old_epoch: u64,
-    pub old_participants: Participants,
-    pub new_participants: Participants,
-    pub threshold: usize,
-    pub public_key: PublicKey,
-    pub protocol: ReshareProtocol,
+    pub contract: ResharingContractState,
+    pub local_private_share: Option<SecretKeyShare>,
+    pub phase: ResharingPhase,
+    pub ready_nonce: u64,
+}
 
+pub struct ResharingReadyState {
+    pub ready: BTreeSet<Participant>,
+    /// Interval to control broadcasting readiness messages.
+    // NOTE: this is an Instant for now since generating/resharing tasks are not async
+    // and happen in main protocol loop. once it becomes async we can make this an interval.
+    pub broadcast_interval: Instant,
+}
+
+pub struct ResharingRunningState {
+    pub protocol: ReshareProtocol,
     /// If the resharing state fails to store data after generating, it gets temporarily
     /// stored here and retried later.
     pub failed_store: Option<SecretKeyShare>,
+    pub started_at: Instant,
+    pub last_activity: Instant,
 }
 
+#[allow(clippy::large_enum_variant)]
+pub enum ResharingPhase {
+    AwaitingReady(ResharingReadyState),
```

Comment:
Would it be possible to rename it to  ResharingAwaitingState? `Ready` in its context sounds like "we are ready for resharing".